exec-analysis/src/main/java/com/capitalone/dashboard/exec/collector/AuditResultCollector.java

```diff
@@ -0,0 +1,116 @@
+package com.capitalone.dashboard.exec.collector;
+
+import com.capitalone.dashboard.exec.model.MetricType;
+import com.capitalone.dashboard.exec.model.HygieiaSparkQuery;
+import com.capitalone.dashboard.exec.model.MetricCollectionStrategy;
+import com.capitalone.dashboard.exec.model.CollectorItemMetricDetail;
+import com.capitalone.dashboard.exec.model.MetricCount;
+import com.capitalone.dashboard.exec.model.CollectorType;
+import com.capitalone.dashboard.exec.repository.PortfolioMetricRepository;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.catalyst.expressions.GenericRowWithSchema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+
+import java.util.Date;
+import java.util.List;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+@Component
+public class AuditResultCollector extends DefaultMetricCollector {
+    private static final String STR_AUDIT_RESULTS = "audit_results";
+    private static final String STR_AUTOMATED = "Automated";
+    private static final String STR_MANUAL = "Manual";
+    private static final String STR_TIME_WINDOW = "timeWindow";
+    private static final String STR_TRACEABILITY = "traceability";
+    private static final String STR_PERCENT_TRACEABILITY = "percentTraceability";
+    private static final Logger LOGGER = LoggerFactory.getLogger(AuditResultCollector.class);
+
+    public AuditResultCollector(PortfolioMetricRepository portfolioMetricRepository) {
+        super(portfolioMetricRepository);
+    }
+
+    @Override
+    protected MetricType getMetricType() {
+        return MetricType.TRACEABILITY;
+    }
+
+    @Override
+    protected String getQuery() {
+        return HygieiaSparkQuery.TRACEABILITY_QUERY_ALL_COLLECTOR_ITEMS;
+    }
+
+    @Override
+    protected String getCollection() {
+        return STR_AUDIT_RESULTS;
+    }
+
+    @Override
+    protected MetricCollectionStrategy getCollectionStrategy() {
+        return MetricCollectionStrategy.LATEST;
+    }
+
+    @Override
+    protected CollectorItemMetricDetail getCollectorItemMetricDetail(List<Row> rowList, MetricType metricType) {
+        CollectorItemMetricDetail collectorItemMetricDetail = new CollectorItemMetricDetail();
+        if (CollectionUtils.isEmpty(rowList)) {
+            return collectorItemMetricDetail;
+        }
+        collectorItemMetricDetail.setType(getMetricType());
+        for (Row row : rowList) {
+            updateCollectorItemMetricDetail(collectorItemMetricDetail, row);
+        }
+        return collectorItemMetricDetail;
+    }
+
+    private void updateCollectorItemMetricDetail(CollectorItemMetricDetail collectorItemMetricDetail,Row itemRow){
```

Comment:
A comment explaining the end result of this update would be best

Reply:
comment added